edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.CloudProxy/CloudConnectionProvider.cs

```diff
@@ -78,6 +81,8 @@ public async Task<Try<ICloudConnection>> Connect(IClientCredentials clientCreden
                 Events.CreatingCloudConnectionUsingClientCredentials(clientCredentials);
                 var cloudListener = new CloudListener(this.edgeHub.Expect(() => new InvalidOperationException("EdgeHub reference should not be null")), clientCredentials.Identity.Id);
                 string productInfo = await this.productInfoStore.GetEdgeProductInfo(clientCredentials.Identity.Id);
+                string modelId = await this.modelIdStore.GetModelId(clientCredentials.Identity.Id);
+                Option<string> modelIdOption = !string.IsNullOrEmpty(modelId) ? Option.Some(modelId) : Option.None<string>();
```

Comment:
>!string.IsNullOrEmpty(modelId) ? Option.Some(modelId) : Option.None<string>(); [](start = 47, length = 78)

Option.Maybe(modelId)

Reply:
see other comment in ModelIdStore, it's prob better to just have it return an Option<string> instead.

---
In reply to: [454164249](https://github.com/Azure/iotedge/pull/3216#discussion_r454164249) [](ancestors = 454164249)